test/spec/modules/essensBidAdapter_spec.js

```diff
@@ -0,0 +1,796 @@
+      expect(payloadJson.imp[2].sizes[0]).to.equal('500x510');
+      expect(payloadJson.imp[2].sizes[1]).to.equal('600x610');
+      sinon.assert.calledOnce(stubLoadScript);
+    });
+    it('should fill all parameters', function () {
+      const bidderRequest = {
+        bidderCode: 'essens',
+        requestId: 'impression-1',
+        bidderRequestId: 'impression-for-essens-1',
+        bids: [
+          {
+            bidId: 'placement1-for_essensT3',
+            bidder: 'essens',
+            requestId: 'essens-impression-1',
+            params: {
+              placementId: 'placement1',
+              dealId: '1234',
+              floorPrice: '23.478'
+            },
+            sizes: [
+              [100, 110],
+              [200, 210]
+            ],
+            placementCode: 'div-media1-top_banner-1',
+            bidderRequestId: 'impression-for-essens-1',
+          }
+        ]
+      }
+
+      const essensadapter = adapter.createNew();
+      essensadapter.callBids(bidderRequest);
+
+      const url = stubLoadScript.getCall(0).args[0];
+      const payload = decodeURIComponent(url.split('&bid=')[1]);
+      const payloadJson = JSON.parse(payload);
+
+      expect(payloadJson.ua).to.exist.and.to.be.a('string');
+      expect(payloadJson.url).to.exist.and.to.be.a('string');
+      expect(Object.keys(payloadJson.imp).length).to.equal(1);
+      expect(payloadJson.imp[0].impressionId).to.equal('placement1-for_essensT3');
+      expect(payloadJson.imp[0].placementId).to.equal('placement1');
+      expect(Object.keys(payloadJson.imp[0].sizes).length).to.equal(2);
+      expect(payloadJson.imp[0].sizes[0]).to.equal('100x110');
+      expect(payloadJson.imp[0].sizes[1]).to.equal('200x210');
+      expect(payloadJson.imp[0].deal).to.equal('1234');
+      expect(payloadJson.imp[0].floorPrice).to.equal('23.478');
+    });
+    it('invalid request: missing mandatory parameters', function () {
+      const bidderRequest = {
+        bidderCode: 'essens',
+        requestId: 'impression-1',
+        bidderRequestId: 'impression-for-essens-1',
+        bids: [
+          {
+            bidId: 'placement1-for_essensT4',
+            bidder: 'essens',
+            requestId: 'essens-impression-1',
+            params: {},
+            sizes: [
+              [100, 110],
+              [200, 210]
+            ],
+            placementCode: 'div-media1-top_banner-1',
+            bidderRequestId: 'impression-for-essens-1',
+          }
+        ]
+      }
+
+      const essensadapter = adapter.createNew();
+      essensadapter.callBids(bidderRequest);
+
+      sinon.assert.notCalled(stubLoadScript);
+    });
+  });
+
+  describe('Test essensResponseHandler method', function () {
+    let stubAddBidResponse
+    beforeEach(() => {
+      stubAddBidResponse = sinon.stub(bidmanager, 'addBidResponse');
+    });
+
+    afterEach(() => {
+      stubAddBidResponse.restore()
+      pbjs._bidsRequested.pop()
```

Comment:
Shouldn't be reliant on prebid private vars. When you stub `addBidRespone` you can grab the bids.  This is important as it can mess with other tests that may be doing this same thing (but shouldn't be).  In fact, since you are stubbing `addBidResponse`

Also should be `$$PREBID_GLOBAL$$` instead of `pbjs`